ee/clickhouse/views/person.py

```diff
@@ -65,26 +70,28 @@ def calculate_funnel_persons(self, request: Request) -> Dict[str, Tuple[list, Op
             return {"result": ([], None, None)}
 
         filter = Filter(request=request, data={"insight": INSIGHT_FUNNELS}, team=self.team)
+        if not filter.limit:
+            filter = filter.with_data({LIMIT: 100})
 
-        funnel_class: Callable
+        funnel_actor_class: Callable
 
         if filter.funnel_viz_type == FunnelVizType.TRENDS:
-            funnel_class = ClickhouseFunnelTrendsPersons
+            funnel_actor_class = ClickhouseFunnelTrendsActors
         else:
             if filter.funnel_order_type == "unordered":
-                funnel_class = ClickhouseFunnelUnorderedPersons
+                funnel_actor_class = ClickhouseFunnelUnorderedActors
             elif filter.funnel_order_type == "strict":
-                funnel_class = ClickhouseFunnelStrictPersons
+                funnel_actor_class = ClickhouseFunnelStrictActors
             else:
-                funnel_class = ClickhouseFunnelPersons
+                funnel_actor_class = ClickhouseFunnelActors
 
-        people, should_paginate = funnel_class(filter, self.team).run()
-        limit = filter.limit if filter.limit else 100
-        next_url = format_query_params_absolute_url(request, filter.offset + limit) if should_paginate else None
+        _, actors = funnel_actor_class(filter, self.team).get_actors()
```

Comment:
hmm, doesn't this run into the same problem we had earlier?

Since `_` >= `actors` always, we should be using `_` for limit / pagination calculation? (like we were doing earlier)

Reply:
_ and actors should be the same under this implementation. The limiting happens strictly at the query level now

Reply:
Ahh, misread things. Ok, they should be the same, legit!

But if they are the same: our pagination logic gets completely borked. The deleted persons pagination test should (would) fail in this case?

(Say there are 150 people, 30 of the first 100 are deleted). Our persons will return 70 (because 30 deleted), without a load more button, while they should return 70, with a load more button that returns the remaining 50.

Reply:
ah the deleted users yeah so the two results wouldn't technically be the same. Should use `_`. Good catch